src/Glpi/Form/QuestionType/QuestionTypeItemDropdown.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Form\QuestionType;
+
+use Dropdown;
+use Override;
+use Session;
+
+final class QuestionTypeItemDropdown extends QuestionTypeItem
+{
+    #[Override]
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->itemtype_aria_label = __('Select a dropdown type');
+        $this->items_id_aria_label = __('Select a dropdown item');
+    }
+
+    public function getAllowedItemtypes(): array
+    {
+        $dropdown_itemtypes = Dropdown::getStandardDropdownItemTypes();
+        array_walk_recursive($dropdown_itemtypes, function (&$value, $key) {
+            $value = $key;
+        });
+
+        return $dropdown_itemtypes;
```

Comment:
Why not returning `array_keys($dropdown_itemtypes)` directly?

Reply:
The `array_keys` function doesn't seem to work for nested arrays.
In concrete terms, I end up with an array containing only the first-level keys, in my case the category headings.

Reply:
My bad, I did not saw it was a recursive walk.